src/pykeen/stoppers/early_stopping.py

```diff
@@ -116,20 +107,15 @@ def __post_init__(self):
         if self.evaluation_triples_factory is None:
             raise ValueError('Must specify a validation_triples_factory or a dataset for using early stopping.')
 
-        if self.larger_is_better:
-            self.improvement_criterion = larger_than_any_buffer_element
-        else:
-            self.improvement_criterion = smaller_than_any_buffer_element
-
-        self.buffer = numpy.empty(shape=(self.patience,))
+        self.remaining_patience = self.patience
 
         # Dummy result tracker
         if self.result_tracker is None:
             self.result_tracker = ResultTracker()
 
     def should_evaluate(self, epoch: int) -> bool:
         """Decide if evaluation should be done based on the current epoch and the internal frequency."""
-        return 0 == ((epoch - 1) % self.frequency)
+        return epoch > 0 and epoch % self.frequency == 0
```

Comment:
looking back, it the old code is really crazy. I don't know why I did it like this